### PR TITLE
Add a Dockerfile

### LINKED ISSUE #14: Add a Dockerfile
- [ ] Create the Dockerfile. See https://github.com/mozilla-services/kinto-dist/blob/master/Dockerfile for example.
- [ ] Configure circle.ci to build a docker image on PR and tags. See https://github.com/mozilla-services/kinto-dist/blob/master/circle.yml

### PR DESCRIPTION
**Pull request by Natim**
Fixes #14

### CODE REVIEW

File: circle.yml [hunk 1]
```diff
@@ -0,0 +1,70 @@
+# These environment variables must be set in CircleCI UI
```
Comment (Natim) [inline circle.yml]: @jasonthomas could you help me setup the CircleCI build for this one?

File: circle.yml [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    # day to keep the cache size down.
+    - I="image-$(date +%j).gz"; if [[ -e ~/docker/$I ]]; then echo "Loading $I"; pigz -d -c ~/docker/$I | docker load; fi
+
+    # create a version.json
+    - >
+        printf '{"name":"%s","commit":"%s","version":"%s","source":"https://github.com/%s/%s","build":"%s"}\n'
+        "kinto-dist"
+        "$CIRCLE_SHA1"
+        "$CIRCLE_TAG"
+        "mozilla-services"
```
Comment (Natim) [inline circle.yml]: Should be Mozilla

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,29 @@
+# Mozilla PollBot server
+FROM python:3.6-slim
+MAINTAINER Product Delivery irc://irc.mozilla.org/#product-delivery
+
+RUN groupadd -g 9000 pollbot && \
```
Comment (mostlygeek) [inline Dockerfile]: UID/GID should be 10001

File: circle.yml [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+        "$CIRCLE_TAG"
+        "mozilla"
+        "$CIRCLE_PROJECT_REPONAME"
+        "$CIRCLE_BUILD_URL"
+        > version.json
+
+    # build the actual deployment container
+    - docker build --pull -t app .
+
+    # write the sha256 sum to an artifact to make image verification easier
```
Comment (mostlygeek) [inline circle.yml]: Remove this output. The only useful SHA sum for verifying a container is only available *after* we push it to docker hub.

File: circle.yml [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+    - docker images --no-trunc | awk '/^app/ {print $3}' | tee $CIRCLE_ARTIFACTS/docker-image-shasum256.txt
+
+    # Clean up any old images; save the new one.
+    - I="image-$(date +%j).gz"; mkdir -p ~/docker; rm ~/docker/*; docker save app | pigz --fast -c > ~/docker/$I; ls -l ~/docker
+
+test:
+  override:
+    -  'true'
```
Comment (mostlygeek) [inline circle.yml]: We should verify that the container works here. At a minimum just run the unit tests in the container.

Comment (Natim) [inline circle.yml]: Do you know how I can run the `make tests-once` command inside the container in the `/app` folder here?

Comment (Natim) [inline circle.yml]: I did `docker run -it -d pollbot make tests-once`

But the thing is we don't have the dev dependencies in the container.

Comment (Natim) [inline circle.yml]: Apparently I could create a script that would install everything in /tmp and run the tests. Is that what you had in mind?

Comment (mostlygeek) [inline circle.yml]: I took a look at the Makefile. It is tailored to use a virtual environment which is unnecessary when testing inside a docker container. 

A pattern that I've encouraged is to set the `ENTRYPOINT` to a shell script. Here is the one I made for the [server-syncstorage](https://github.com/mozilla-services/server-syncstorage/blob/master/docker-entrypoint.sh). 

Here's how tests are run in server-syncstorage's [circle.yml](https://github.com/mozilla-services/server-syncstorage/blob/master/circle.yml): 

```yaml
test:
  override:
  - docker run -it app:build test_flake8
  - docker run -it app:build test_nose
  - docker run -it app:build test_functional
```

Comment (Natim) [inline circle.yml]: Seems very nice I can try to do that indeed.

File: Dockerfile [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+    apt-get install -y --no-install-recommends $buildDeps && \
+    pip install -e /app && \
+    # cleanup
+    apt-get purge -y $buildDeps && \
+    rm -rf /var/lib/apt/lists/*
+
+USER pollbot
+
+# Start the pollbot server
+CMD pollbot
```
Comment (mostlygeek) [inline Dockerfile]: Suggestiong: If you use a shell script as an `ENTRYPOINT` you can keep the testing logic in one place. Here's server-syncstorage as an example: 

* [ENTRYPOINT](https://github.com/mozilla-services/server-syncstorage/blob/master/docker-entrypoint.sh) 
* [Dockerfile](https://github.com/mozilla-services/server-syncstorage/blob/master/Dockerfile#L9-L11)
* [usage in circle.yml](https://github.com/mozilla-services/server-syncstorage/blob/master/circle.yml#L34-L38)

testing in circle.yml
```yaml
test:
  override:
  - docker run -it app:build test_flake8
  - docker run -it app:build test_nose
  - docker run -it app:build test_functional
```

File: circle.yml (hunk 1 above)
Comment (jasonthomas) [inline circle.yml]: Done

File: Dockerfile (hunk 6 above)
Comment (Natim) [inline Dockerfile]: It is a great idea, I have created #44 to address this.

**Pull request merged**